tools/clang/lib/Parse/HLSLRootSignature.cpp

```diff
@@ -476,14 +483,22 @@ bool RootSignatureTokenizer::IsAlpha(char c) const
 
 RootSignatureParser::RootSignatureParser(
     RootSignatureTokenizer *pTokenizer, DxilRootSignatureVersion DefaultVersion,
-    llvm::raw_ostream &OS)
-    : m_pTokenizer(pTokenizer), m_Version(DefaultVersion), m_OS(OS) {}
+    DxilRootSignatureCompilationFlags CompilationFlags, llvm::raw_ostream &OS)
+    : m_pTokenizer(pTokenizer), m_Version(DefaultVersion), m_CompilationFlags(CompilationFlags), m_OS(OS) {}
 
 HRESULT RootSignatureParser::Parse(DxilVersionedRootSignatureDesc **ppRootSignature)
 {
     HRESULT hr = S_OK;
     DxilVersionedRootSignatureDesc *pRS = NULL;
 
+    DXASSERT(!((bool)(m_CompilationFlags & DxilRootSignatureCompilationFlags::GlobalRootSignature) && 
+               (bool)(m_CompilationFlags & DxilRootSignatureCompilationFlags::LocalRootSignature)),
+             "global and local cannot be both set");
+
+    if (m_CompilationFlags == DxilRootSignatureCompilationFlags::None) {
+      m_CompilationFlags = DxilRootSignatureCompilationFlags::GlobalRootSignature;
```

Comment:
We don't want to force the flag in the parser.  Just make sure the places in the compiler that parse root signatures pass the global flag, except for the LocalRootSignature subobject which passes the local flag.  Otherwise this code can't be used in a local/global agnostic way.  In fact, there's one code path that we want to allow either one: see comment above on FrontendActions.cpp.